sympy/functions/special/bessel.py

```diff
@@ -685,14 +704,173 @@ class yn(SphericalBesselBase):
     def _rewrite(self):
         return self._eval_rewrite_as_bessely(self.order, self.argument)
 
+    def _eval_rewrite_as_besselj(self, nu, z):
+        return (-1)**(nu+1) * sqrt(pi/(2*z)) * besselj(-nu - S.Half, z)
+
+    def _eval_rewrite_as_bessely(self, nu, z):
```

Comment:
@asmeurer do you think that I have to assert that ``nu`` is an Integer or a symbolic Integer?